validphys2/src/validphys/core.py

```diff
@@ -801,6 +801,11 @@ def get_description(self):
         dbpath = self.path.parent/'theory.db'
         return fetch_theory(dbpath, self.id)
 
+    def get_eko(self):
+        if self.is_pineappl():
+            return self.path/'eko.tar'
+        raise TypeError(f"the requested eko does not exists because the theory {self.id} is not a pineappl theory")
```

Comment:
I would remove this `get_eko` function and try to get the `eko.tar` regardless of whether it is a pineappl theory